src/controller/OpenRobertaController.vala

```diff
@@ -0,0 +1,238 @@
+/*
+ * brickman -- Brick Manager for LEGO MINDSTORMS EV3/ev3dev
+ *
+ * Copyright 2015 Stefan Sauer <ensonic@google.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+/* OpenRobertaController.vala - Controller for openroberta lab service */
+
+using Ev3devKit.Ui;
+
+namespace BrickManager {
+    // TODO: move to lib/openroberta/Manager.vala
+    [DBus (name = "org.openroberta.lab")]
+    interface OpenRobertaLab : Object {
+        [DBus (name = "status")]
+        public signal void status (string message);
+
+        [DBus (name = "connect")]
+        public abstract string connect (string address) throws IOError;
+        [DBus (name = "disconnect")]
+        public abstract void disconnect () throws IOError;
+    }
+
+    public class OpenRobertaController : Object, IBrickManagerModule {
+        const string SERVICE_NAME = "org.openroberta.lab";
+        const string CONFIG = "/etc/openroberta.conf";
+
+        OpenRobertaWindow open_roberta_window;
+        internal OpenRobertaStatusBarItem status_bar_item;
+        OpenRobertaLab service;
+        KeyFile config;
+        MessageDialog pin_dialog;
+        bool executing_user_code = false;
+
+        public bool available { get; set; default = false; }
+
+        public string display_name { get { return "Open Roberta Lab"; } }
+
+        public void show_main_window () {
+            open_roberta_window.show ();
+        }
+
+        public OpenRobertaController () {
+            // TODO: defer window creation to show_main_window()
+            open_roberta_window = new OpenRobertaWindow (display_name);
+            status_bar_item = new OpenRobertaStatusBarItem ();
+            config = new KeyFile ();
+
+            bind_property ("available", status_bar_item, "visible", BindingFlags.SYNC_CREATE);
+            bind_property ("available", open_roberta_window.menu, "visible", BindingFlags.SYNC_CREATE);
+
+            notify["available"].connect (() => {
+                if (!available) {
+                    open_roberta_window.info.text =
```

Comment:
It seems like if we are setting this here, we would need to set it to something else in the "else" block.